atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraNamespaceDeleterTest.java

```diff
@@ -0,0 +1,186 @@
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleter;
+import java.util.List;
+import java.util.Map;
+import org.apache.cassandra.thrift.KsDef;
+import org.apache.cassandra.thrift.NotFoundException;
+import org.apache.cassandra.thrift.TimedOutException;
+import org.apache.thrift.TException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class CassandraNamespaceDeleterTest {
+    private static final String KEYSPACE = "nonreservedkeyspace";
+    private static final ImmutableCqlQuery DROP_KEYSPACE_QUERY = CqlQuery.builder()
+            .safeQueryFormat(SchemaBuilder.dropKeyspace("\"" + KEYSPACE + "\"")
+                    .ifExists()
+                    .buildInternal())
+            .build();
+
+    @Mock
+    private CassandraClient cassandraClient;
+
+    private NamespaceDeleter namespaceDeleter;
+
+    @Before
+    public void before() {
+        namespaceDeleter = new CassandraNamespaceDeleter(getConfig(), () -> cassandraClient);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsKeyspaceSpecifiedInConfig() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient).execute_cql3_query(eq(DROP_KEYSPACE_QUERY), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsOnlyKeyspaceSpecified() throws TException {
+        setupCassandraSchemaVersions();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        verify(cassandraClient, times(1)).execute_cql3_query(any(), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceWaitsForSchemaVersionsBeforeDroppingKeyspace() throws TException {
+        when(cassandraClient.describe_schema_versions()).thenReturn(Map.of("hello", List.of(), "world", List.of()));
+
+        assertThatThrownBy(namespaceDeleter::deleteAllDataFromNamespace)
+                .hasCauseInstanceOf(IllegalStateException.class)
+                .cause()
+                .hasMessageContaining("Cassandra cluster cannot come to agreement on schema versions");
+        verify(cassandraClient, never()).execute_cql3_query(any(), any(), any());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceWaitsForSchemaVersionsAfterDroppingKeyspace() throws TException {
+        when(cassandraClient.describe_schema_versions())
+                .thenReturn(Map.of("hello", List.of("world")))
+                .thenReturn(Map.of("hello", List.of(), "world", List.of()));
+
+        assertThatThrownBy(namespaceDeleter::deleteAllDataFromNamespace)
+                .hasCauseInstanceOf(IllegalStateException.class)
+                .cause()
+                .hasMessageContaining("Cassandra cluster cannot come to agreement on schema versions");
```

Comment:
sure, it'll throw, but we've already dropped it...

what happens in this case (in prod code)? Concretely it looks like it'll just throw, but I'm curious how we'll handle this case.

Reply:
Nothing really happens - it's more just to show that we're doing the wait for schema versions as asked for by internal cass team. If it throws, it either failed to get to agreement before or after the call.
If before, we'll just retry :/
If after, we've already dropped, and when we retry it'll be a no-op.

In truth, we don't strictly _need_ to wait for agreement, it's to stop cass from being overloaded by a bunch of schema changes in succession